diff -urN at91bootstrap-2.13-rc4-0rig/board/at91sam9g45ek/at91sam9g45ek_defconfig at91bootstrap-2.13-rc4/board/at91sam9g45ek/at91sam9g45ek_defconfig
--- at91bootstrap-2.13-rc4-0rig/board/at91sam9g45ek/at91sam9g45ek_defconfig	2009-08-22 17:03:08.000000000 +0200
+++ at91bootstrap-2.13-rc4/board/at91sam9g45ek/at91sam9g45ek_defconfig	2009-08-22 20:35:17.000000000 +0200
@@ -1,6 +1,6 @@
 #
 # Automatically generated make config: don't edit
-# Fri Aug 21 18:57:33 2009
+# Sat Aug 22 20:24:56 2009
 #
 HAVE_DOT_CONFIG=y
 CONFIG_BOARDNAME="at91sam9g45ek"
@@ -101,6 +101,7 @@
 CONFIG_DEBUG=y
 CONFIG_VERBOSE=y
 CONFIG_HW_INIT=y
+CONFIG_TRAP_HANDLER=y
 # CONFIG_USER_HW_INIT is not set
 # CONFIG_THUMB is not set
 CONFIG_SX_AT91=y
diff -urN at91bootstrap-2.13-rc4-0rig/Config.in at91bootstrap-2.13-rc4/Config.in
--- at91bootstrap-2.13-rc4-0rig/Config.in	2009-08-22 17:03:08.000000000 +0200
+++ at91bootstrap-2.13-rc4/Config.in	2009-08-22 20:22:02.000000000 +0200
@@ -112,6 +112,13 @@
 	help
 	  Initialize Hardware
 
+
+config CONFIG_TRAP_HANDLER
+	bool "Trap Handler"
+	default y	if !(CONFIG_AT91SAM9260EK || CONFIG_AFEB9260)
+	help
+	  Initialize Hardware
+
 config CONFIG_USER_HW_INIT
 	bool "Call User specific Hardware Initialization"
 	help
diff -urN at91bootstrap-2.13-rc4-0rig/crt0_gnu.S at91bootstrap-2.13-rc4/crt0_gnu.S
--- at91bootstrap-2.13-rc4-0rig/crt0_gnu.S	2009-08-22 17:03:08.000000000 +0200
+++ at91bootstrap-2.13-rc4/crt0_gnu.S	2009-08-22 20:24:15.000000000 +0200
@@ -35,6 +35,7 @@
 	.text
 
  	#include "include/part.h"
+ 	#include "include/trap.h"
 
 /*----------------------------------------------------------------------------
  Area Definition
@@ -60,6 +61,50 @@
 	b 	irq_vector	/* IRQ : read the AIC */
 	b 	fiq_vector      /* FIQ */
 
+#if defined(CONFIG_TRAP_HANDLER)
+	.globl trap_handler
+fiq_vector:
+	ldr 	r0, =TRAP_FIQ
+	ldr     r4, = trap_handler
+	mov     lr, pc
+	bx      r4
+
+undef_vector:
+	ldr 	r0, =TRAP_UNDEF
+	ldr     r4, = trap_handler
+	mov     lr, pc
+	bx      r4
+
+swi_vector:
+	ldr 	r0, =TRAP_SWI
+	ldr     r4, = trap_handler
+	mov     lr, pc
+	bx      r4
+
+pabt_vector:
+	ldr 	r0, =TRAP_PABT
+	ldr     r4, = trap_handler
+	mov     lr, pc
+	bx      r4
+
+dabt_vector:
+	ldr 	r0, =TRAP_DABT
+	ldr     r4, = trap_handler
+	mov     lr, pc
+	bx      r4
+
+rsvd_vector:
+	ldr 	r0, =TRAP_RSVD
+	ldr     r4, = trap_handler
+	mov     lr, pc
+	bx      r4
+
+irq_vector:
+	ldr 	r0, =TRAP_IRQ
+	ldr     r4, = trap_handler
+	mov     lr, pc
+	bx      r4
+#else
 undef_vector:
 	b 	undef_vector
 swi_vector:
@@ -73,7 +118,8 @@
 irq_vector:
 	b 	irq_vector
 fiq_vector:
-	b 	fiq_vector
+	b	fiq_vector
+#endif
 reset_vector:
 
 /* Init the stack */
diff -urN at91bootstrap-2.13-rc4-0rig/driver/driver.mk at91bootstrap-2.13-rc4/driver/driver.mk
--- at91bootstrap-2.13-rc4-0rig/driver/driver.mk	2009-08-22 17:03:08.000000000 +0200
+++ at91bootstrap-2.13-rc4/driver/driver.mk	2009-08-22 20:22:41.000000000 +0200
@@ -7,6 +7,8 @@
 COBJS-y				+=	$(DRIVERS)debug.o
 COBJS-y				+=	$(DRIVERS)gpio.o
 COBJS-y				+=	$(DRIVERS)pmc.o
+COBJS-$(CONFIG_TRAP_HANDLER)	+=	$(DRIVERS)trap.o
+
 
 COBJS-$(CONFIG_SDRAM)		+=	$(DRIVERS)sdramc.o
 COBJS-$(CONFIG_SDDRC)		+=	$(DRIVERS)sddrc.o
diff -urN at91bootstrap-2.13-rc4-0rig/driver/trap.c at91bootstrap-2.13-rc4/driver/trap.c
--- at91bootstrap-2.13-rc4-0rig/driver/trap.c	1970-01-01 01:00:00.000000000 +0100
+++ at91bootstrap-2.13-rc4/driver/trap.c	2009-08-22 19:22:55.000000000 +0200
@@ -0,0 +1,60 @@
+/* ----------------------------------------------------------------------------
+ *         ATMEL Microcontroller Software Support  -  ROUSSET  -
+ * ----------------------------------------------------------------------------
+ * Copyright (c) 2006, Atmel Corporation
+
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * - Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the disclaimer below.
+ *
+ * Atmel's name may not be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * DISCLAIMER: THIS SOFTWARE IS PROVIDED BY ATMEL "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT ARE
+ * DISCLAIMED. IN NO EVENT SHALL ATMEL BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+ * OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+ * EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ * ----------------------------------------------------------------------------
+ * File Name           : trap.c
+ * Object              :
+ * Creation            : USa Aug 22th 2009
+ *-----------------------------------------------------------------------------
+ */
+#include "part.h"
+#include "main.h"
+#include "debug.h"
+#include "trap.h"
+
+typedef	char	string[6];
+
+string	trap_message[8] = {
+	"RESET",
+	"UNDEF",
+	"SWI",
+	"PABT",
+	"DABT",
+	"RSVD",
+	"IRQ",
+	"FIQ"
+};
+	
+void	trap_handler(unsigned int i)
+{
+	dbg_print("\r\n");
+	dbg_print("TRAP: ");
+	dbg_print(trap_message[i]);	
+	dbg_print("\r\n");	
+	asm("mov	r0, #0");
+	asm("bx	r0");
+}
+
diff -urN at91bootstrap-2.13-rc4-0rig/include/trap.h at91bootstrap-2.13-rc4/include/trap.h
--- at91bootstrap-2.13-rc4-0rig/include/trap.h	1970-01-01 01:00:00.000000000 +0100
+++ at91bootstrap-2.13-rc4/include/trap.h	2009-08-22 18:18:42.000000000 +0200
@@ -0,0 +1,45 @@
+/* ----------------------------------------------------------------------------
+ *         ATMEL Microcontroller Software Support  -  ROUSSET  -
+ * ----------------------------------------------------------------------------
+ * Copyright (c) 2009, Atmel Corporation
+
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * - Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the disclaiimer below.
+ *
+ * Atmel's name may not be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * DISCLAIMER: THIS SOFTWARE IS PROVIDED BY ATMEL "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT ARE
+ * DISCLAIMED. IN NO EVENT SHALL ATMEL BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+ * OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+ * EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ * ----------------------------------------------------------------------------
+ * File Name           : trap.h
+ * Object              : ATMEL ARM Exception Ids
+ * Creation            : USa 2009-02-10
+ *---------------------------------------------------------------------------
+ */
+#ifndef _TRAP_H
+#define _TRAP_H
+
+#define	TRAP_RESET	0
+#define	TRAP_UNDEF	1
+#define	TRAP_SWI	2
+#define	TRAP_PABT	3
+#define	TRAP_DABT	4
+#define	TRAP_RSVD	5
+#define	TRAP_IRQ	6
+#define	TRAP_FIQ	7
+
+#endif
diff -urN at91bootstrap-2.13-rc4-0rig/Makefile at91bootstrap-2.13-rc4/Makefile
--- at91bootstrap-2.13-rc4-0rig/Makefile	2009-08-22 17:03:08.000000000 +0200
+++ at91bootstrap-2.13-rc4/Makefile	2009-08-22 18:40:14.000000000 +0200
@@ -356,6 +356,9 @@
 $(obj)nandflash.o: $(DRIVERS_SRC)/nandflash.c .config
 	$(CC)	$(CPPFLAGS) -c -o $(obj)nandflash.o $(DRIVERS_SRC)/nandflash.c
 
+$(obj)trap.o:	$(DRIVERS_SRC)/trap.c .config
+	$(CC)	$(CPPFLAGS) -c -o $(obj)trap.o $(DRIVERS_SRC)/trap.c
+
 ../$(TOOLCHAIN_DIR):
 	(cd .. ; git clone git://git.busybox.net/~ulf/git/$(TOOLCHAIN_DIR))
 
