diff -urN u-boot-2009.08-rc1-0rig/Makefile u-boot-2009.08-rc1/Makefile
--- u-boot-2009.08-rc1-0rig/Makefile	2009-07-27 10:37:37.000000000 +0200
+++ u-boot-2009.08-rc1/Makefile	2009-07-31 22:09:32.000000000 +0200
@@ -20,6 +20,7 @@
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston,
 # MA 02111-1307 USA
 #
+BUILD_DATE= $(shell date +%F)
 
 VERSION = 2009
 PATCHLEVEL = 08
@@ -285,6 +286,22 @@
 # Always append ALL so that arch config.mk's can add custom ones
 ALL += $(obj)u-boot.srec $(obj)u-boot.bin $(obj)System.map $(U_BOOT_NAND) $(U_BOOT_ONENAND)
 
+ifneq ($(BOARDNAME),)
+BINARY := u-boot-$(BOARDNAME)-$(U_BOOT_VERSION)-$(REVISION)
+BINARY_BIN := $(obj)$(BINARY).bin
+BINARY_GZ  := $(obj)$(BINARY).gz
+ALL += $(BINARY_BIN)
+ALL += $(BINARY_GZ)
+else
+BINARY := u-boot
+BINARY_BIN := $(obj)$(BINARY).bin
+BINARY_GZ  := $(obj)$(BINARY).gz
+endif
+
+ifeq ($(DESTDIR),)
+DESTDIR=binaries
+endif
+
 all:		$(ALL)
 
 $(obj)u-boot.hex:	$(obj)u-boot
@@ -293,9 +310,20 @@
 $(obj)u-boot.srec:	$(obj)u-boot
 		$(OBJCOPY) -O srec $< $@
 
-$(obj)u-boot.bin:	$(obj)u-boot
+$(BINARY_BIN):	$(obj)u-boot
 		$(OBJCOPY) ${OBJCFLAGS} -O binary $< $@
 
+$(BINARY_GZ):	$(BINARY_BIN)
+		gzip -c $(BINARY_BIN) > $(BINARY_GZ)
+
+install:	all
+		-install -d	$(DESTDIR)
+		install		$(BINARY_BIN)	$(DESTDIR)
+
+tftp:		install
+		cp $(BINARY_GZ)				$(TFTPBOOT)/$(BINARY).gz
+		cp $(BINARY_BIN) 			$(TFTPBOOT)/$(BINARY).bin
+
 $(obj)u-boot.ldr:	$(obj)u-boot
 		$(obj)tools/envcrc --binary > $(obj)env-ldr.o
 		$(LDR) -T $(CONFIG_BFIN_CPU) -c $@ $< $(LDR_FLAGS)
@@ -3657,6 +3685,7 @@
 		-o -name '*.srec' -o -name '*.bin' -o -name u-boot.img \) \
 		-print0 \
 		| xargs -0 rm -f
+	@rm -f $(obj)u-boot-*.gz
 	@rm -f $(OBJS) $(obj)*.bak $(obj)ctags $(obj)etags $(obj)TAGS \
 		$(obj)cscope.* $(obj)*.*~
 	@rm -f $(obj)u-boot $(obj)u-boot.map $(obj)u-boot.hex $(ALL)

