diff -urN u-boot-2009.08-rc2-0rig/Makefile u-boot-2009.08-rc2/Makefile
--- u-boot-2009.08-rc2-0rig/Makefile	2009-08-20 11:06:24.000000000 +0200
+++ u-boot-2009.08-rc2/Makefile	2009-08-20 11:22:18.000000000 +0200
@@ -2838,28 +2838,51 @@
 pm9261_config	:	unconfig
 	@$(MKCONFIG) $(@:_config=) arm arm926ejs pm9261 ronetix at91
 
+at91sam9g45ek_nandflash_config \
+at91sam9g45ek_dataflash_config \
+at91sam9g45ek_dataflash_cs0_config \
+at91sam9g45ek_eeprom_config \
+at91sam9g45ek_config \
+at91sam9m10ek_nandflash_config \
+at91sam9m10ek_dataflash_config \
+at91sam9m10ek_dataflash_cs0_config \
+at91sam9m10ek_eeprom_config \
+at91sam9m10ek_config \
 at91sam9m10g45ek_nandflash_config \
 at91sam9m10g45ek_dataflash_config \
 at91sam9m10g45ek_dataflash_cs0_config \
 at91sam9m10g45ek_config \
+at91sam9m10g45ek_eeprom_config \
 at91sam9g45ekes_nandflash_config \
 at91sam9g45ekes_dataflash_config \
 at91sam9g45ekes_dataflash_cs0_config \
+at91sam9g45ekes_eeprom_config	\
 at91sam9g45ekes_config	:	unconfig
 	@mkdir -p $(obj)include
-		@if [ "$(findstring 9m10,$@)" ] ; then \
-		echo "#define CONFIG_AT91SAM9M10G45EK 1"	>>$(obj)include/config.h ; \
-		$(XECHO) "... 9M10G45 Variant" ; \
-	else \
-		echo "#define CONFIG_AT91SAM9G45EKES 1"	>>$(obj)include/config.h ; \
+	@if [ "$(findstring 9m10g45ek,$@)" ] ; then \
+		echo "#define CONFIG_AT91SAM9M10G45EK 1"	>>$(obj)include/config.h ; \
+		$(XECHO) "... AT91SAM9M10G45EK Variant" ; \
+	elif [ "$(findstring 9m10ek,$@)" ] ; then \
+		echo "#define CONFIG_AT91SAM9M10G45EK 1"	>>$(obj)include/config.h ; \
+		echo "#define CONFIG_AT91SAM9M10EK 1"		>>$(obj)include/config.h ; \
+		$(XECHO) "... AT91SAM9M10EK Variant" ; \
+	elif [ "$(findstring 9m10ek,$@)" ] ; then \
+		echo "#define CONFIG_AT91SAM9M10G45EK 1"	>>$(obj)include/config.h ; \
+		echo "#define CONFIG_AT91SAM9G45EK 1"		>>$(obj)include/config.h ; \
+		$(XECHO) "... AT91SAM9G45EK Variant" ; \
+	elif [ "$(findstring 9m10g45ek,$@)" ] ; then \
+		echo "#define CONFIG_AT91SAM9G45EKES 1"		>>$(obj)include/config.h ; \
+		$(XECHO) "... AT91SAM9G45EKES Variant" ; \
 	fi;
-
 	@if [ "$(findstring _nandflash,$@)" ] ; then \
 		echo "#define CONFIG_SYS_USE_NANDFLASH 1"	>>$(obj)include/config.h ; \
 		$(XECHO) "... with environment variable in NAND FLASH" ; \
-	else \
+	elif [ "$(findstring _dataflash,$@)" ] ; then \
 		echo "#define CONFIG_ATMEL_SPI 1"	>>$(obj)include/config.h ; \
 		$(XECHO) "... with environment variable in SPI DATAFLASH CS0" ; \
+	elif [ "$(findstring _eeprom,$@)" ] ; then \
+		echo "#define CONFIG_ATMEL_EEPROM 1"		>>$(obj)include/config.h ; \
+		$(XECHO) "... with environment variable in SPI EEPROM (not yet supported)" ; \
 	fi;
 	@$(MKCONFIG) -a at91sam9m10g45ek arm arm926ejs at91sam9m10g45ek atmel at91
 
