diff -urN u-boot-2009.08-rc2-0rig/common/comment.c u-boot-2009.08-rc2/common/comment.c
--- u-boot-2009.08-rc2-0rig/common/comment.c	1970-01-01 01:00:00.000000000 +0100
+++ u-boot-2009.08-rc2/common/comment.c	2009-08-22 09:22:23.000000000 +0200
@@ -0,0 +1,54 @@
+/*
+ * (C) Copyright 2000
+ * Wolfgang Denk, DENX Software Engineering, wd@denx.de.
+ *
+ * See file CREDITS for list of people who contributed to this
+ * project.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of
+ * the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ * MA 02111-1307 USA
+ */
+
+/*
+ * Boot support
+ */
+#include <common.h>
+#include <command.h>
+
+
+/* 
+ * This allows you to have comments starting with "#" in
+ * a file downloaded to U-Boot.
+ * If this is not implemented, then if your file
+ * has a comment, then U-Boot will write an error message.
+ * While the error message is sent out, u-boot will not
+ * listen to incoming data over the UART, and if the linedelay
+ * is not set high enough, then you will miss characters.
+ * If the linedelay is high enough, then the download will be
+ * longer
+ */
+int do_comment (cmd_tbl_t *cmdtp, int flag, int argc, char *argv[])
+{
+	int	i;
+	/* Ignore, but no error message is printed */
+	return 0;
+}
+
+U_BOOT_CMD(
+	#,	1,	1,	do_comment,
+	"#\t- comment\n",
+	"\n"
+);
+
diff -urN u-boot-2009.08-rc2-0rig/common/Makefile u-boot-2009.08-rc2/common/Makefile
--- u-boot-2009.08-rc2-0rig/common/Makefile	2009-08-22 09:16:27.000000000 +0200
+++ u-boot-2009.08-rc2/common/Makefile	2009-08-22 09:23:53.000000000 +0200
@@ -49,6 +49,7 @@
 
 # environment
 COBJS-y += env_common.o
+COBJS-y += comment.o
 COBJS-$(CONFIG_ENV_IS_IN_DATAFLASH) += env_dataflash.o
 COBJS-$(CONFIG_ENV_IS_IN_EEPROM) += env_eeprom.o
 COBJS-$(CONFIG_ENV_IS_EMBEDDED) += env_embedded.o
